File: CS/DataContext_Ex/Class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;

namespace DataContext_Ex {

    class Person : DependencyObject {
        public string FirstName {
            get { return (string)GetValue(FirstNameProperty); }
            set { SetValue(FirstNameProperty, value); }
        }

        public string LastName {
            get { return (string)GetValue(LastNameProperty); }
            set { SetValue(LastNameProperty, value); }
        }

        public string Title {
            get { return (string)GetValue(TitleProperty); }
            set { SetValue(TitleProperty, value); }
        }

        public string Country {
            get { return (string)GetValue(CountryProperty); }
            set { SetValue(CountryProperty, value); }
        }

        public string City {
            get { return (string)GetValue(CityProperty); }
            set { SetValue(CityProperty, value); }
        }

        public string HomePhone {
            get { return (string)GetValue(HomePhoneProperty); }
            set { SetValue(HomePhoneProperty, value); }
        }

        public static readonly DependencyProperty FirstNameProperty =
            DependencyProperty.Register("FirstName", typeof(string), typeof(Person), new UIPropertyMetadata(string.Empty));

        public static readonly DependencyProperty LastNameProperty =
            DependencyProperty.Register("LastName", typeof(string), typeof(Person), new UIPropertyMetadata(string.Empty));

        public static readonly DependencyProperty TitleProperty =
            DependencyProperty.Register("Title", typeof(string), typeof(Person), new UIPropertyMetadata(string.Empty));

        public static readonly DependencyProperty CountryProperty =
            DependencyProperty.Register("Country", typeof(string), typeof(Person), new UIPropertyMetadata(string.Empty));

        public static readonly DependencyProperty CityProperty =
            DependencyProperty.Register("City", typeof(string), typeof(Person), new UIPropertyMetadata(string.Empty));
        
        public static readonly DependencyProperty HomePhoneProperty =
            DependencyProperty.Register("HomePhone", typeof(string), typeof(Person), new UIPropertyMetadata(string.Empty));
    }


    class MyDescriptionClass : DependencyObject {
        public string Description {
            get { return (string)GetValue(DescriptionProperty); }
            set { SetValue(DescriptionProperty, value); }
        }

        public static readonly DependencyProperty DescriptionProperty =
            DependencyProperty.Register("Description", typeof(string), typeof(MyDescriptionClass), new UIPropertyMetadata(string.Empty));

    }
}

File: VB/DataContext_Ex/Class1.vb
Imports System.Windows

Namespace DataContext_Ex

    Friend Class Person
        Inherits DependencyObject

        Public Property FirstName As String
            Get
                Return CStr(GetValue(FirstNameProperty))
            End Get

            Set(ByVal value As String)
                SetValue(FirstNameProperty, value)
            End Set
        End Property

        Public Property LastName As String
            Get
                Return CStr(GetValue(LastNameProperty))
            End Get

            Set(ByVal value As String)
                SetValue(LastNameProperty, value)
            End Set
        End Property

        Public Property Title As String
            Get
                Return CStr(GetValue(TitleProperty))
            End Get

            Set(ByVal value As String)
                SetValue(TitleProperty, value)
            End Set
        End Property

        Public Property Country As String
            Get
                Return CStr(GetValue(CountryProperty))
            End Get

            Set(ByVal value As String)
                SetValue(CountryProperty, value)
            End Set
        End Property

        Public Property City As String
            Get
                Return CStr(GetValue(CityProperty))
            End Get

            Set(ByVal value As String)
                SetValue(CityProperty, value)
            End Set
        End Property

        Public Property HomePhone As String
            Get
                Return CStr(GetValue(HomePhoneProperty))
            End Get

            Set(ByVal value As String)
                SetValue(HomePhoneProperty, value)
            End Set
        End Property

        Public Shared ReadOnly FirstNameProperty As DependencyProperty = DependencyProperty.Register("FirstName", GetType(String), GetType(Person), New UIPropertyMetadata(String.Empty))

        Public Shared ReadOnly LastNameProperty As DependencyProperty = DependencyProperty.Register("LastName", GetType(String), GetType(Person), New UIPropertyMetadata(String.Empty))

        Public Shared ReadOnly TitleProperty As DependencyProperty = DependencyProperty.Register("Title", GetType(String), GetType(Person), New UIPropertyMetadata(String.Empty))

        Public Shared ReadOnly CountryProperty As DependencyProperty = DependencyProperty.Register("Country", GetType(String), GetType(Person), New UIPropertyMetadata(String.Empty))

        Public Shared ReadOnly CityProperty As DependencyProperty = DependencyProperty.Register("City", GetType(String), GetType(Person), New UIPropertyMetadata(String.Empty))

        Public Shared ReadOnly HomePhoneProperty As DependencyProperty = DependencyProperty.Register("HomePhone", GetType(String), GetType(Person), New UIPropertyMetadata(String.Empty))
    End Class

    Friend Class MyDescriptionClass
        Inherits DependencyObject

        Public Property Description As String
            Get
                Return CStr(GetValue(DescriptionProperty))
            End Get

            Set(ByVal value As String)
                SetValue(DescriptionProperty, value)
            End Set
        End Property

        Public Shared ReadOnly DescriptionProperty As DependencyProperty = DependencyProperty.Register("Description", GetType(String), GetType(MyDescriptionClass), New UIPropertyMetadata(String.Empty))
    End Class
End Namespace
